privacyidea/lib/resolvers/LDAPIdResolver.py

```diff
@@ -764,15 +767,25 @@ def loadConfig(self, config):
         self.resolverId = self.uri
         self.authtype = config.get("AUTHTYPE", AUTHTYPE.SIMPLE)
         self.tls_verify = is_true(config.get("TLS_VERIFY", False))
-        # Fallback to TLSv1. (int: 3, TLSv1.1: 4, v1.2: 5)
-        self.tls_version = int(config.get("TLS_VERSION") or ssl.PROTOCOL_TLSv1)
-
+        # Fallback to TLSv1. (int: 3, TLSv1.1: 4, v1.2: 5, TLS negotiation: 2)
+        self.tls_version = int(config.get("TLS_VERSION") or DEFAULT_TLS_PROTOCOL)
+        if self.tls_version:
+            tls_version = self.tls_version
+            tls_options = TLS_OPTIONS_1_3 if tls_version == int(ssl.PROTOCOL_TLS) else None
+        else:
+            tls_version = int(DEFAULT_TLS_PROTOCOL)
+            tls_options = None
         self.tls_ca_file = config.get("TLS_CA_FILE") or DEFAULT_CA_FILE
-        if self.tls_verify and (self.uri.lower().startswith("ldaps") or
-                                    self.start_tls):
-            self.tls_context = Tls(validate=ssl.CERT_REQUIRED,
-                                   version=self.tls_version,
-                                   ca_certs_file=self.tls_ca_file)
+        if self.uri.lower().startswith("ldaps") or self.start_tls:
+            if self.tls_verify:
+                self.tls_context = Tls(validate=ssl.CERT_REQUIRED,
+                                       version=tls_version,
+                                       ssl_options=tls_options,
+                                       ca_certs_file=self.tls_ca_file)
+            else:
+                self.tls_context = Tls(validate=ssl.CERT_NONE,
+                                       version=tls_version,
+                                       ssl_options=tls_options)
```

Comment:
Do I understand it correctly, that we now need a TLS context, even if the certificate is not verified, because TLS 1.3 does not work without a TLS context?